Creating a materialized view of rollup partition causes FE crash

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. CREATE TABLE orders ( 
    dt date NOT NULL, 
    order_id bigint NOT NULL, 
    user_id int NOT NULL, 
    merchant_id int NOT NULL, 
    good_id int NOT NULL, 
    good_name string NOT NULL, 
    price int NOT NULL, 
    cnt int NOT NULL, 
    revenue int NOT NULL, 
    state tinyint NOT NULL 
) 
PRIMARY KEY (dt, order_id) 
PARTITION BY RANGE(`dt`) 
(
PARTITION p20210820 VALUES [('2021-08-20'), ('2021-08-21')), 
PARTITION p20210821 VALUES [('2021-08-21'), ('2021-08-22'))
) 
DISTRIBUTED BY HASH(order_id) BUCKETS 4
PROPERTIES (
    "replication_num" = "1", 
    "enable_persistent_index" = "true"
);
2. CREATE MATERIALIZED VIEW order_mv2
PARTITION BY date_trunc('MONTH', `dt`)
DISTRIBUTED BY HASH(`order_id`) BUCKETS 10
REFRESH ASYNC START('2023-07-01 10:00:00') EVERY (interval 1 day)
AS
select
    dt,
    order_id,
    user_id,
    sum(cnt) as total_cnt,
    sum(revenue) as total_revenue, 
    count(state) as state_count
from orders
group by dt, order_id, user_id;

### Expected behavior (Required)
Create Success.
### Real behavior (Required)
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10094
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:1011) ~[classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1938) [classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1889) [classes/:?]
        at com.starrocks.leader.Checkpoint.replayAndGenerateGlobalStateMgrImage(Checkpoint.java:215) [classes/:?]
        at com.starrocks.leader.Checkpoint.runAfterCatalogReady(Checkpoint.java:106) [classes/:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:73) [classes/:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [classes/:?]
Caused by: com.starrocks.sql.analyzer.SemanticException: date_trunc function can't support argument other than year|quarter|month|week|day
        at com.starrocks.sql.analyzer.FunctionAnalyzer.analyze(FunctionAnalyzer.java:60) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitFunctionCall(ExpressionAnalyzer.java:897) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitFunctionCall(ExpressionAnalyzer.java:214) ~[classes/:?]
        at com.starrocks.analysis.FunctionCallExpr.accept(FunctionCallExpr.java:454) ~[classes/:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:55) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.bottomUpAnalyze(ExpressionAnalyzer.java:211) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:111) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyzeExpression(ExpressionAnalyzer.java:1217) ~[classes/:?]
        at com.starrocks.catalog.MaterializedView.analyzePartitionInfo(MaterializedView.java:612) ~[classes/:?]
        at com.starrocks.catalog.MaterializedView.onCreate(MaterializedView.java:578) ~[classes/:?]
        at com.starrocks.catalog.Database.createMaterializedWithLock(Database.java:551) ~[classes/:?]
        at com.starrocks.server.LocalMetastore.replayCreateMaterializedView(LocalMetastore.java:2701) ~[classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayCreateMaterializedView(GlobalStateMgr.java:2594) ~[classes/:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:260) ~[classes/:?]
        ... 6 more
2023-02-06 17:12:53,993 WARN (leaderCheckpointer|84) [GlobalStateMgr.replayJournal():1891] got interrupt exception or inconsistent exception when replay journal 523335, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10094
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:1011) ~[classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1938) ~[classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1889) [classes/:?]
        at com.starrocks.leader.Checkpoint.replayAndGenerateGlobalStateMgrImage(Checkpoint.java:215) [classes/:?]
        at com.starrocks.leader.Checkpoint.runAfterCatalogReady(Checkpoint.java:106) [classes/:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:73) [classes/:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [classes/:?]
Caused by: com.starrocks.sql.analyzer.SemanticException: date_trunc function can't support argument other than year|quarter|month|week|day
        at com.starrocks.sql.analyzer.FunctionAnalyzer.analyze(FunctionAnalyzer.java:60) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitFunctionCall(ExpressionAnalyzer.java:897) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer$Visitor.visitFunctionCall(ExpressionAnalyzer.java:214) ~[classes/:?]
        at com.starrocks.analysis.FunctionCallExpr.accept(FunctionCallExpr.java:454) ~[classes/:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:55) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.bottomUpAnalyze(ExpressionAnalyzer.java:211) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:111) ~[classes/:?]
        at com.starrocks.sql.analyzer.ExpressionAnalyzer.analyzeExpression(ExpressionAnalyzer.java:1217) ~[classes/:?]
        at com.starrocks.catalog.MaterializedView.analyzePartitionInfo(MaterializedView.java:612) ~[classes/:?]
        at com.starrocks.catalog.MaterializedView.onCreate(MaterializedView.java:578) ~[classes/:?]
        at com.starrocks.catalog.Database.createMaterializedWithLock(Database.java:551) ~[classes/:?]
        at com.starrocks.server.LocalMetastore.replayCreateMaterializedView(LocalMetastore.java:2701) ~[classes/:?]
        at com.starrocks.server.GlobalStateMgr.replayCreateMaterializedView(GlobalStateMgr.java:2594) ~[classes/:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:260) ~[classes/:?]
        ... 6 more
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.5.0

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java
@@ -939,7 +939,7 @@ public class MaterializedView extends OlapTable implements GsonPostProcessable {
             return SyncPartitionUtils.calcSyncSamePartition(basePartitionMap, mvPartitionMap);
         } else if (partitionExpr instanceof FunctionCallExpr) {
             FunctionCallExpr functionCallExpr = (FunctionCallExpr) partitionExpr;
-            String granularity = ((StringLiteral) functionCallExpr.getChild(0)).getValue();
+            String granularity = ((StringLiteral) functionCallExpr.getChild(0)).getValue().toLowerCase();
             return SyncPartitionUtils.calcSyncRollupPartition(basePartitionMap, mvPartitionMap,
                     granularity, partitionColumn.getPrimitiveType());
         } else {

--- a/fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java
+++ b/fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java
@@ -403,7 +403,7 @@ public class PartitionBasedMaterializedViewRefreshProcessor extends BaseTaskRunP
                 partitionDiff = SyncPartitionUtils.calcSyncSamePartition(basePartitionMap, mvPartitionMap);
             } else if (partitionExpr instanceof FunctionCallExpr) {
                 FunctionCallExpr functionCallExpr = (FunctionCallExpr) partitionExpr;
-                String granularity = ((StringLiteral) functionCallExpr.getChild(0)).getValue();
+                String granularity = ((StringLiteral) functionCallExpr.getChild(0)).getValue().toLowerCase();
                 partitionDiff = SyncPartitionUtils.calcSyncRollupPartition(basePartitionMap, mvPartitionMap,
                         granularity, partitionColumn.getPrimitiveType());
             }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java
@@ -45,18 +45,22 @@ public class FunctionAnalyzer {
             if (!(functionCallExpr.getChild(0) instanceof StringLiteral)) {
                 throw new SemanticException("date_trunc requires first parameter must be a string constant");
             }
-            final StringLiteral fmtLiteral = (StringLiteral) functionCallExpr.getChild(0);
+
+            // The uppercase parameters stored in the metadata are converted to lowercase parameters,
+            // because BE does not support uppercase parameters
+            final String lowerParam = ((StringLiteral) functionCallExpr.getChild(0)).getValue().toLowerCase();
+            functionCallExpr.setChild(0, new StringLiteral(lowerParam));
 
             if (functionCallExpr.getChild(1).getType().isDatetime()) {
 
                 if (!Lists.newArrayList("year", "quarter", "month", "week", "day", "hour", "minute", "second")
-                        .contains(fmtLiteral.getStringValue())) {
+                        .contains(lowerParam)) {
                     throw new SemanticException("date_trunc function can't support argument other than " +
                             "year|quarter|month|week|day|hour|minute|second");
                 }
             } else if (functionCallExpr.getChild(1).getType().isDate()) {
                 if (!Lists.newArrayList("year", "quarter", "month", "week", "day")
-                        .contains(fmtLiteral.getStringValue())) {
+                        .contains(lowerParam)) {
                     throw new SemanticException("date_trunc function can't support argument other than " +
                             "year|quarter|month|week|day");
                 }

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/SelectStmtTest.java
@@ -42,6 +42,7 @@ import com.starrocks.planner.ProjectNode;
 import com.starrocks.planner.UnionNode;
 import com.starrocks.qe.ConnectContext;
 import com.starrocks.qe.ShowResultSet;
+import com.starrocks.sql.ast.DropTableStmt;
 import com.starrocks.sql.optimizer.operator.scalar.CastOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
 import com.starrocks.sql.plan.ExecPlan;
@@ -168,6 +169,13 @@ public class SelectStmtTest {
         starRocksAssert.query(sql).explainQuery();
         sql = "select localtimestamp";
         starRocksAssert.query(sql).explainQuery();
+    }
+
+    @Test
+    public void testDateTruncUpperCase() throws Exception {
+        String sql = "select date_trunc('MONTH', CAST('2020-11-04 11:12:13' AS DATE));";
+        ConnectContext ctx = starRocksAssert.getCtx();
+        UtFrameUtils.parseStmtWithNewParser(sql, ctx);
     }
 
     @Test

